lib/compute/jenkins-main-node.ts

```diff
@@ -368,71 +377,31 @@ export class JenkinsMainNode {
-        // eslint-disable-next-line max-len
-        + `${JenkinsMainNodeConfig.adminRolePermissions().map((e) => ` -s /hudson/authorizationStrategy/roleMap[2]/role/permissions -t elem -n "permission" -v ${e}`).join(' ')}`
-        + ' -s /hudson/authorizationStrategy/roleMap[2]/role -t elem -n "assignedSIDs" -v " " '
-        // eslint-disable-next-line max-len
-        + `${this.admins(oidcFederateProps.adminUsers).map(((e) => ` -s /hudson/authorizationStrategy/roleMap[2]/role/assignedSIDs -t elem -n "sid" -v ${e}`)).join(' ')}`
-        + ' -s /hudson/authorizationStrategy --type elem -n roleMap'
-        + ' -i /hudson/authorizationStrategy/roleMap[3] -t attr -n "type" -v "slaveRolesRoles"'
-        + ' -s /hudson/authorizationStrategy/roleMap[2] -t elem -n role -v " "'
-        + ' -i /hudson/authorizationStrategy/roleMap[2]/role[2] -t attr -n "name" -v "read-only"'
-        + ' -i /hudson/authorizationStrategy/roleMap[2]/role[2] -t attr -n "pattern" -v ".*"'
-        + ' -s /hudson/authorizationStrategy/roleMap[2]/role[2] -t elem -n permissions -v " "'
-        // eslint-disable-next-line max-len
-        + `${JenkinsMainNodeConfig.readOnlyRolePermissions().map((e) => ` -s /hudson/authorizationStrategy/roleMap[2]/role[2]/permissions -t elem -n "permission" -v ${e}`).join(' ')}`
-        + ' -s /hudson/authorizationStrategy/roleMap[2]/role[2] -t elem -n "assignedSIDs" -v " "'
-        + ' -s /hudson/authorizationStrategy/roleMap[2]/role[2]/assignedSIDs -t elem -n "sid" -v "anonymous"'
-        + ' /var/lib/jenkins/config.xml'
-        + ` && ${RELOAD_CONFIG}`
-        : 'echo OIDC disabled: Not enabling Role Based Authentication'),
-
-      // sleep for 30s before editing config.xml again
-      InitCommand.shellCommand('sleep 30'),
-
-      // Enabling OIDC
       InitCommand.shellCommand(oidcFederateProps.runWithOidc
         // eslint-disable-next-line max-len
         ? `var=\`aws --region ${stackRegion} secretsmanager get-secret-value --secret-id ${oidcFederateProps.oidcCredArn} --query SecretString --output text\` && `
-        + ' xmlstarlet ed -L -d "/hudson/securityRealm"'
-        + ' -s /hudson -t elem -n securityRealm -v " "'
-        + ' -i //securityRealm -t attr -n "class" -v "org.jenkinsci.plugins.oic.OicSecurityRealm"'
-        + ' -i //securityRealm -t attr -n "plugin" -v "oic-auth@1.8"'
+        + ' varkeys=`echo $var | yq \'keys\' | cut -d "-" -f2 | cut -d " " -f2` &&'
         // eslint-disable-next-line max-len
-        + `${JenkinsMainNodeConfig.oidcConfigFields().map((e) => ` -s /hudson/securityRealm -t elem -n ${e[0]} -v ${e[1] === 'replace' ? `"$(echo $var | jq -r ".${e[0]}")"` : `"${e[1]}"`}`).join(' ')}`
-        + ' /var/lib/jenkins/config.xml'
-        + ` && ${RELOAD_CONFIG}`
+        + ' for i in $varkeys; do newvalue=`echo $var | yq .$i` && myenv=$newvalue i=$i yq -i \'.jenkins.securityRealm.oic.[env(i)]=env(myenv)\' /var/lib/jenkins/jenkins.yaml ; done'
```

Comment:
Shouldn't we parse the yml file instead? How is this different from xmlstarlet?

Reply:
Please refer the comment above. we are not adding any config via this command, Just replacing with user provided values

Reply:
This is exactly what we can do with xml as well right?

Reply:
We are adding configuration via code (yaml in this case). You can check linked issues for details why we reached this decision. Also editing, maintaining and manipulating yaml is much easier than xml. Now on, every user does not have to learn xmlstartlet to add code to xml. It was a very messy process and changes in hudson libraries and config might have affected us. Right now, yaml you add the config you want in yaml and yaml takes care of it to add it to config. xml.

TL;DR:
Easy to maintain, codeify and update the config via code. (It was like we were editing bytecodes by editing config.xml which will have serious implications and unknown errors.)

Reply:
Do we have to learn `yq` now?

Reply:
Just for replacing any values in jenkins.yaml on the jenkins server directly during installation.